File: playbook-java/src/main/java/io/infinitic/playbook/java/scheduler/Start.java
package io.infinitic.playbook.java.scheduler;

import io.infinitic.clients.InfiniticClient;
import io.infinitic.playbook.java.scheduler.workflows.RecurringWorkflowInput;
import io.infinitic.playbook.java.scheduler.workflows.RecurringWorkflowScheduler;

import java.util.Set;

public class Start {
    static final String schedulerTag = "scheduler";

    public static void main(String[] args) {
        try(InfiniticClient client = InfiniticClient.fromConfigResource("/infinitic.yml")) {
            // create a stub for RecurringWorkflowScheduler, with tag "scheduler"
            RecurringWorkflowScheduler workflow = client.newWorkflow(RecurringWorkflowScheduler.class, Set.of(schedulerTag));

            // start new workflow
            client.dispatchVoid(workflow::schedule, "* * * * *", new RecurringWorkflowInput());
        }
    }
}

File: playbook-kotlin/src/main/kotlin/io/infinitic/playbook/kotlin/scheduler/Start.kt
package io.infinitic.playbook.kotlin.scheduler

import io.infinitic.clients.InfiniticClient
import io.infinitic.playbook.kotlin.scheduler.workflows.RecurringWorkflowInput
import io.infinitic.playbook.kotlin.scheduler.workflows.RecurringWorkflowScheduler
import java.time.format.DateTimeFormatter

internal const val schedulerTag = "scheduler"

fun main() {
    InfiniticClient.fromConfigResource("/infinitic.yml").use { client ->
        // create a stub for PromisesWorkflow, with tag "scheduler"
        val scheduler = client.newWorkflow(RecurringWorkflowScheduler::class.java, setOf(schedulerTag))

        // start new workflow
        client.dispatch(scheduler::schedule, "* * * * *", RecurringWorkflowInput())
    }
}
